### PR TITLE
Update to use the latest site.

### LINKED ISSUE #57: Update to use Terra-site v2
# Issue Description
Update to use Terra Site v2 to pull in the master branches of terra-core, terra-clinical, and terra-framework.

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Lets get terra-ui up and running!!

This PR:
- Adds terra-core, framework and clinical components
- Adds translations
- Updates/Adds the Home and ReadMe sections

Resolves #57

### CODE REVIEW

**Comment by emilyrohrbough** [comment]
Note:
- When https://github.com/cerner/terra-site/pull/17 is merged, when navigation to a repository's component list, the repo's homepage will display before a component is selected.
- When https://github.com/cerner/terra-site/pull/21 is merged, the github logo should display with a link to all the terra repos

**Comment by emilyrohrbough** [comment]
@bjankord @mjhenkes @ryanthemanuel @JakeLaCombe This PR is ready for review. This has been updated to have the newest translations, use React 16 and the latest version of site.

**Comment by emilyrohrbough** [comment]
On second thought, maybe we should wait for the first terra-dev-site cut?

**Comment by mjhenkes** [comment]
We should get PR deploys setup for this repo

File: .travis.yml [hunk 1]
```diff
@@ -1,22 +1,20 @@
 language: node_js
+cache: false
```
Comment (mjhenkes) [inline .travis.yml]: Good call here, we should try out removing the cache from our other repos too.

Comment (mjhenkes) [reply .travis.yml]: Though I wonder if we could just delete the line instead of explicitly setting it to false.

Comment (mjhenkes) [reply .travis.yml]: Looks like it doesn't hurt anything.
https://docs.travis-ci.com/user/caching/#Explicitly-disabling-caching

File: .travis.yml [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
-#   skip_cleanup: true
-#   github_token: $GITHUB_TOKEN # Set in travis-ci.org dashboard
-#   local_dir: build
-#   on:
-#     branch: master
+before_install:
+  - docker pull selenium/standalone-chrome:3.8.1-erbium
+  - docker run -v /dev/shm:/dev/shm -e TZ=America/Chicago -d -p 4444:4444 selenium/standalone-chrome:3.8.1-erbium
+before_deploy:
+  - lerna run build:production
```
Comment (mjhenkes) [inline .travis.yml]: should this be npm run build:production instead? This isn't a mono repo is it?

Comment (emilyrohrbough) [reply .travis.yml]: good call

File: .travis.yml [hunk 3]
```diff
@@ -1,22 +1,20 @@
...
-# deploy:
-#   provider: pages
-#   skip_cleanup: true
-#   github_token: $GITHUB_TOKEN # Set in travis-ci.org dashboard
-#   local_dir: build
-#   on:
-#     branch: master
+before_install:
+  - docker pull selenium/standalone-chrome:3.8.1-erbium
+  - docker run -v /dev/shm:/dev/shm -e TZ=America/Chicago -d -p 4444:4444 selenium/standalone-chrome:3.8.1-erbium
```
Comment (mjhenkes) [inline .travis.yml]: Do we have wdio tests in terra-ui.com? I wonder, do we need them? If we don't, we could remove this docker stuff here.

Comment (bjankord) [reply .travis.yml]: I dont think we need them in this repo.

File: .travis.yml [hunk 4]
```diff
@@ -1,22 +1,17 @@
...
-#   github_token: $GITHUB_TOKEN # Set in travis-ci.org dashboard
-#   local_dir: build
-#   on:
-#     branch: master
+before_deploy:
+  - npm run build:production
+deploy:
+  provider: pages
+  skip_cleanup: true
+  github_token: $GITHUB_TOKEN # Set in travis-ci.org dashboard
```
Comment (mjhenkes) [inline .travis.yml]: Reminder mostly for myself, We need to setup this token for gh-pages deploy.

File: package.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "terra-ui_com",
+  "name": "terra-ui.com",
```
Comment (mjhenkes) [inline package.json]: If anyone is curious like I was, `.` is allowed in package names. https://docs.npmjs.com/files/package.json

File: package.json [hunk 6]
```diff
@@ -19,11 +19,13 @@
     "clean": "node scripts/clean-modules/index.js",
     "clean:install": "npm run clean && npm install",
     "build:test": "NODE_ENV=test webpack --config webpack.config.js",
-    "build:production": "NODE_ENV=production webpack --config webpack.config.js -p",
+    "build:production": "NODE_ENV=production webpack --config webpack.config.prod.js -p",
```
Comment (mjhenkes) [inline package.json]: why don't we reverse `config` and `prod` so its `webpack.prod.config.js`

File: site.config.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
+import 'terra-core/packages/terra-site/src/cerner-mock-theme.scss';
+
+import navConfig from './navigation.config';
+// import componentConfig from './generatedComponentConfig';
```
Comment (mjhenkes) [inline site.config.js]: Delete comment?

File: generatedComponentConfig.js [hunk 8]
```diff
@@ -0,0 +1,915 @@
+/* This file is auto-generated by terra-site's generate-component-config config. */
```
Comment (bjankord) [inline generatedComponentConfig.js]: We should add this file to the .gitignore

File: .gitignore [hunk 9]
```diff
@@ -23,4 +23,4 @@ node_modules
 # Generated js bundles
 /public/webpack/*
```
Comment (bjankord) [inline .gitignore]: We should add `generatedComponentConfig.js` to this file

File: webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
+const webpackConfig = require('terra-dev-site/src/config/webpack.config');
+
+const babelLoderRule = {
```
Comment (bjankord) [inline webpack.config.js]: nit: `Loder` => `Loader`

File: src/Home/HomeMoreInfo.scss [hunk 11]
```diff
@@ -5,7 +5,7 @@
     padding: 4rem 1.429rem;
     text-align: center;
   }
-
+
```
Comment (bjankord) [inline src/Home/HomeMoreInfo.scss]: fry_squinting

File: .travis.yml (hunk 4 above)
Comment (mjhenkes) [reply .travis.yml]: This key has been added.

File: .gitignore (hunk 9 above)
Comment (emilyrohrbough) [reply .gitignore]: If we add the generateComponentConfig, it cannot be referenced within the site config, which then loads it into site.

File: package.json (hunk 5 above)
Comment (emilyrohrbough) [reply package.json]: the '_' broke the generate-config pattern which caused the change initially

File: package.json [hunk 12]
```diff
@@ -19,12 +19,15 @@
     "clean": "node scripts/clean-modules/index.js",
     "clean:install": "npm run clean && npm install",
     "build:test": "NODE_ENV=test webpack --config webpack.config.js",
-    "build:production": "NODE_ENV=production webpack --config webpack.config.js -p",
+    "build:production": "NODE_ENV=production node --max_old_space_size=2000 node_modules/webpack/bin/webpack.js --config webpack.prod.config.js -p",
```
Comment (mjhenkes) [inline package.json]: webpack was running out of memory optimizing for prod. I'd like to do more investigation here, but for the time being i bumped up the heap size for the process.

File: site.config.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+const siteConfig = {
+  /* The navigation configuration.  */
+  navConfig,
+
+  /* The component configuration path. */
+  componentConfig,
+
+  /* The logo to display as the placeholder. */
+  placeholderSrc: 'https://github.com/cerner/terra-core/raw/master/terra.png',
```
Comment (JakeLaCombe) [inline site.config.js]: It’d be nice to replace this with an svg at some point. Not sure the exact scope needed, as we may need UX help in it possibly.

File: src/Footer/Index.scss [hunk 14]
```diff
@@ -0,0 +1,16 @@
+:local {
+  .footer {
+    background: #004c76;
```
Comment (JakeLaCombe) [inline src/Footer/Index.scss]: I recommend setting a foreground color here as well to ensure valid contrast. Did the call outs or UX direction have a value used for most text?

File: webpack.config.js [hunk 15]
```diff
@@ -0,0 +1,21 @@
+const webpackConfig = require('terra-dev-site/src/config/webpack.config');
+
+const babelLoaderRule = {
+  test: /\.(jsx|js)$/,
+  exclude: /node_modules(?!\/terra-)/,
+  use: 'babel-loader',
+};
+
+const moduleRules = webpackConfig.module.rules;
+moduleRules.shift();
```
Comment (JakeLaCombe) [inline webpack.config.js]: Can you leave a comment about the module and reason for shifting?

File: src/Footer/Index.scss (hunk 14 above)
Comment (emilyrohrbough) [reply src/Footer/Index.scss]: This is the same styles (copy & pasted) that were applied to the original terra-ui site footer. This is only displayed on the first Home page. I am open to suggestions on improving this! I might be good to scope these changes in a new issue though?

Comment (mjhenkes) [reply src/Footer/Index.scss]: In the interest of getting  a working combined doc site out sooner. I'd like to tackle improving this in another issue.

Comment (JakeLaCombe) [reply src/Footer/Index.scss]: I’m fine with that

Comment (JakeLaCombe) [reply src/Footer/Index.scss]: I’m fine with that

Comment (JakeLaCombe) [reply src/Footer/Index.scss]: I’m fine with that

Comment (JakeLaCombe) [reply src/Footer/Index.scss]: I’m fine with that

Comment (JakeLaCombe) [reply src/Footer/Index.scss]: I’m fine with that

File: .gitignore (hunk 9 above)
Comment (bjankord) [reply .gitignore]: whompwhomp

**Pull request merged**